docs/async_command_pattern.md

```diff
@@ -0,0 +1,82 @@
+# Async Command pattern
+
+The async command pattern allows developing commands where the controller sends a command to the data plane and doesn't expect an immediate response.
+
+This is similar to the request-response, with the important difference that the command execution doesn't block the reconciliation, that is the command may take a while to execute, and in the meantime the reconciler can flag the k8s resource as _processing_ such command.
+
+The flow of the async command pattern looks like:
+
+<!-- PlantUML source:
+
+@startuml
+
+activate Controller
+Controller -> "Data plane" ++: Send AsyncCommand
+
+"Data plane" -> Controller: Ack
+note left: Flag the Resource as processing
+deactivate Controller
+note over Controller: Short-circuit reconciliation
+
+note right of "Data plane": Process the command
+
+"Data plane" -> Controller ++: Send AsyncCommandResult
+
+Controller -> "Data plane": Ack
+note left: Flag the Resource as succeeded/failed
+deactivate "Data plane"
+
+note over Controller: Continue reconciliation
+deactivate Controller
+
+@enduml
+
+-->
+
+![](https://plantuml-server.kkeisuke.dev/svg/ZP7BRW8n34Nt_WgBBKBTpmA1gBgkoXTOZcV6QYP6YOFKls-01GEQFdPUnDVtdEoAK_OwHG1YrpEvuC6IPujHCjn7t6nnzKfEU8gKP8NhTOT7IG7tvIlnmQQ9KW1uUDDsxWaTxlaJahKBKNhly2tIW3uAVaYncbcG2fwoiPIYQO0WIvMk0NPkZURHnz6oRrWpLtNCmfPOevAh9RZjP1r6H-iVC3fylnsy5k6_APQv6q6D3h_u-XzzgSmI9Bpqf572NC4y37wmS9arLNaMi6mITWsZVVqt.svg)
+
+In order to implement this pattern, some utilities are provided:
+
+* `service.NewAsyncCommandHandler` which wraps `ServiceMessage` in order to simplify sending back the async result:
```

Comment:
`ServiceMessage` ... not 100% sold on that name :smile_cat:  but that's more details :-)

Reply:
`ServiceMessage` is comes from `Service` interface, to distinguish from `Message`... If you don't like it, can you open an issue and propose a better name? :)

Reply:
yeah, I saw the service in there later on... all good.

(service still trigger an uncomfortable feeling in me :smile: )

Reply:
@matzew I really suggest you to propose another name then in an issue, those names have to be comfortable for people :smile: